companion.py

```diff
@@ -565,6 +572,9 @@ def query(self, endpoint: str) -> CAPIData:  # noqa: CCR001
 
         try:
             logger.trace_if('capi.query', 'Trying...')
+            if conf_module.capi_pretend_down:
```

Comment:
This is actually CAPI.  However, please add `{endpoint}` to the exception text.  It's that or you'd instead need to do it at the station etc call sites *into* `query()`.

Reply:
It doesn't matter if that results in a very wide GUI window due to it being in the status line, this is only for developer use anyway.

Reply:
In GUI it will show `Frontier CAPI query failure: {endpoint}` anyway, but I've added endpoint to `ServerError` exception's message, no problems.

Reply:
> In GUI it will show `Frontier CAPI query failure: {endpoint}` anyway, but I've added endpoint to `ServerError` exception's message, no problems.

Except this means I now realise the code needs further adjustments.  I do want it to be explicit when it's the "pretend" case, stay tuned for further review comments.